node/libs/concurrency/src/sync/prunable_queue.rs

```diff
@@ -0,0 +1,148 @@
+//! Prunable queue mimics a producer-consumer channel, in enabling
```

Comment:
nit: if you want to present it as mpsc with pruning, maybe "prunable_mpsc" would be more accurate name? Just a suggestion.

Reply:
also document that it is unbounded.

Reply:
I didn't call it a channel because it doesn't cover all the functionality of one, but it might be less confusing option at this point. 

I renamed as `prunable_mpsc`.

Reply:
I didn't call it a channel because it doesn't cover all the functionality of one, but it might be the less confusing option at this point. 

I renamed as `prunable_mpsc`.